.dagger/src/GotenbergBundle.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DaggerModule;
+
+use Dagger\Attribute\DaggerFunction;
+use Dagger\Attribute\DaggerObject;
+use Dagger\Attribute\DefaultPath;
+use Dagger\Attribute\Doc;
+use Dagger\Attribute\ReturnsListOfType;
+use Dagger\Container;
+use Dagger\Directory;
+use Dagger\Service;
+use function Dagger\dag;
+
+#[DaggerObject]
+#[Doc('Module for GotenbergBundle')]
+class GotenbergBundle
+{
+    #[DaggerFunction]
+    #[Doc('Returns a Gotenberg container.')]
+    public function gotenbergContainer(
+        string $gotenbergVersion = '8.0',
+    ): Container {
+        return dag()
+            ->container()
+            ->from("gotenberg/gotenberg:{$gotenbergVersion}")
+        ;
+    }
+
+    #[DaggerFunction]
+    #[Doc('Returns a Gotenberg service.')]
+    public function gotenbergService(
+        string $gotenbergVersion = '8.0',
+    ): Service {
+        return $this->gotenbergContainer($gotenbergVersion)
+            ->withExposedPort(3000)
+            ->asService()
+        ;
+    }
+
+    #[DaggerFunction]
+    #[Doc('Returns a PHP container.')]
+    public function phpContainer(
+        #[DefaultPath('.')]
+        Directory $source,
+
+        string $phpVersion = '8.4',
+    ): Container {
```

Comment:
Correct me if I'm wrong, but this looks like an internal method. If you do not expect anyone to be typing `dagger call php-container` then you can remove the attributes and make it private.

```suggestion
    private function phpContainer(
        Directory $source, 
        string $phpVersion = '8.4'
    ): Container {
```


Dagger only cares what goes in and what comes out. 
Once you're into a DaggerFunction, you can do whatever you want, normal PHP rules apply. You can use private methods, other classes, whatever; as long as you return what it expects at the end.

Reply:
Yes I thought about dagger query and stuff like this where there seems to be a best practice about small atomic functions that can be composed in any way. Maybe this does not apply to "projects" but only reusable functions ?

Reply:
updated to private